### PR TITLE
Feat: EcOp Builtin

### PR DESCRIPTION
**Pull request by rodrigo-pino**
solves #156

### CODE REVIEW

File: pkg/vm/builtins/ecop.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+		inputsFelt[i] = felt
+	}
+
+	// Note: the python vm has an upper limit on the size of `m`(the fifth input)  but
+	// since it is always maxout at 2**252, I see no point on adding a check
+	// for it for now
+
+	// alpha and beta are paremeters required by the curve
+	// extracted from pedersen_params.json in https://github.com/starkware-libs/cairo-lang
+	alpha := f.One()
```
Comment (jmjac) [inline pkg/vm/builtins/ecop.go:77]: Let's move these to the util package cause ecdsa builtin also requires the same values

File: pkg/vm/builtins/ecop.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+	// get the y coordinate: y = slope * (p.X - x) - p.Y
+	y := f.Element{}
+	y.Sub(&p.X, &x)
+	y.Mul(&y, &slope)
+	y.Sub(&y, &p.Y)
+
+	return point{x, y}
+}
+
```
Comment (jmjac) [inline pkg/vm/builtins/ecop.go:194]: Not sure if the comment is correct here. Isn't here only one `y` coordinate here?

File: pkg/vm/builtins/ecop.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+// structure to represent a point in the elliptic curve
+type point struct {
+	X, Y f.Element
+}
+
+// returns true if a point `p` belongs to the `ec` curve ruled by the params `alpha` and
+// `beta`. In other words, true if  y^2 = x^3 + alpha * x + beta
+func (p *point) onCurve(alpha, beta *f.Element) bool {
```
[the remark this answers is not in the document]
Comment (rodrigo-pino) [reply pkg/vm/builtins/ecop.go:112]: Yes, I think that is something to look for. I think we can make an analysis as a separate issue, what do you think? Including benchmark results as well.

File: pkg/vm/builtins/ecop.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+	// get the y coordinate: y = slope * (p.X - x) - p.Y
+	y := f.Element{}
+	y.Sub(&p.X, &x)
+	y.Mul(&y, &slope)
+	y.Sub(&y, &p.Y)
+
+	return point{x, y}
+}
+
+// performs elliptic curve doubling over a point. Assumes `y` coordinates are
```
[the remark this answers is not in the document]
Comment (rodrigo-pino) [reply pkg/vm/builtins/ecop.go:195]: Yes, bad comment

File: pkg/vm/builtins/ecop.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmjac) [reply pkg/vm/builtins/ecop.go:112]: Sounds good, let's just merge this and benchmark in the future

**Review by jmjac — APPROVED** [review]

**Pull request merged**